src/main/java/com/commercetools/helper/mapper/impl/payment/ProductNameMapper.java

```diff
@@ -0,0 +1,93 @@
+package com.commercetools.helper.mapper.impl.payment;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import io.sphere.sdk.carts.LineItem;
+import io.sphere.sdk.models.EnumValue;
+import io.sphere.sdk.models.LocalizedEnumValue;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.products.attributes.Attribute;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import static com.commercetools.payment.constants.Psp.PRODUCT_NAME_MAX_LENGTH;
+
+@Component
+public class ProductNameMapper {
+
+    protected String prefixProductNameWithAttr;
+
+    @Value("${prefixProductNameWithAttr:#{null}}")
+    public void setPrefixProductNameWithAttr(String prefixProductNameWithAttr) {
+        this.prefixProductNameWithAttr = prefixProductNameWithAttr;
+    }
+
+    /**
+     * @param lineItem
+     * @param locales
+     * @return productName String that will be passed to paypal
+     */
+    public String getPaypalItemName(@Nonnull LineItem lineItem, @Nonnull List<Locale> locales) {
+        String productName = lineItem.getName().get(locales);
+        if (StringUtils.isNotBlank(prefixProductNameWithAttr)) {
+            String withPrefix = getPrefixForProductName(lineItem, locales) + productName;
+            return withPrefix.length() < PRODUCT_NAME_MAX_LENGTH ? withPrefix : productName;
+        }
+        return productName;
+    }
+
+    private String getPrefixForProductName(@Nonnull LineItem lineItem, @Nonnull List<Locale> locales) {
+        String prefix = "";
+        ProductVariant variant = lineItem.getVariant();
+        if (Objects.nonNull(variant) && Objects.nonNull(variant.getAttributes()) &&
+                Objects.nonNull(prefixProductNameWithAttr)) {
```

Comment:
`Objects.nonNull(prefixProductNameWithAttr)` should be the first statement in your `if` condition. Also, why do you need to check it here if you ensure its existence here: https://github.com/commercetools/commercetools-paypal-plus-integration/pull/166/files#diff-9e0a94a0c7392524378cf805a0598dd2R40 ?

Reply:
dfadfc990daae968dbee6bc1c84fbe532a9efb1f